### PR TITLE
create CalculateSize()

### PR DESCRIPTION
**Pull request by deitch**
We already pull out the size of the file to use when reporting updates; might as well expose it and simplify our routines.

### CODE REVIEW

**Comment by deitch** [comment]
Hey @ImJasonH and @jonjohnsonjr ; this is the final follow-up from those other ones, notably #675 and its predecessors.

I might start doing some of these progress and other things for the v1layout format as well, once this is in.

File: pkg/v1/tarball/write.go
Comment (ImJasonH) [inline pkg/v1/tarball/write.go:288]: Having both this and [`CalculateTarFileSize`](https://godoc.org/github.com/google/go-containerregistry/pkg/v1/tarball#CalculateTarFileSize) seems confusing. Can we rename one, or make it unexported?

Comment (deitch) [reply pkg/v1/tarball/write.go:288]: Sure, I can rename `CalculateTarFileSize()`. 5 mins.

Comment (deitch) [reply pkg/v1/tarball/write.go:288]: Done. Have a look.

Comment (deitch) [reply pkg/v1/tarball/write.go:288]: Ouch, this is still open. Can we get a review, merge it in, close it out?

File: pkg/v1/tarball/write.go
Comment (jonjohnsonjr) [inline pkg/v1/tarball/write.go:436]: Does this need to be exported?

File: pkg/v1/tarball/write.go
Comment (jonjohnsonjr) [reply pkg/v1/tarball/write.go:288]: Oof, it's been a strange couple months so I've let some things slip, sorry about that.

File: pkg/v1/tarball/write.go
Comment (deitch) [reply pkg/v1/tarball/write.go:436]: Need to? Not really, but it is such a nice convenience function that it might be useful.

You kind of wish this were built into core golang `archive/tar` package.

**Comment by deitch** [comment]
> Oof, it's been a strange couple months so I've let some things slip, sorry about that.

No worries.

**Comment by deitch** [comment]
While we are at it, separately, would you take a "download progress updates" on `Layer()` and similar like we did on the entire tarball?

**Comment by deitch** [comment]
This has been open super long. Can we get it accepted and merged or denied for a reason and closed @ImJasonH or @jonjohnsonjr ? Thanks

File: pkg/v1/tarball/write.go
Comment (jonjohnsonjr) [reply pkg/v1/tarball/write.go:436]: I agree this is useful, but I'd like to avoid exposing anything unless it's really necessary. They might accept a commit to expose some of `archive/tar`? https://github.com/golang/go/blob/8f2db14cd35bbd674cb2988a508306de6655e425/src/archive/tar/format.go#L141-L152

**Review by jonjohnsonjr — COMMENTED** [review]
Going through and cleaning up old issues -- I didn't realize we never merged this. If you'd still like to get it merged, it LGTM other than exposing `CalculateSingleFileInTarSize`, which feels out of place.

File: pkg/v1/tarball/write.go
Comment (deitch) [reply pkg/v1/tarball/write.go:436]: Sure, I will privatize it and rebase and update. Check soon.

Comment (deitch) [reply pkg/v1/tarball/write.go]: I got it. Take another look. It has been a while, I might have made an error on the rebase, so please check carefully. Another pair of eyes is always helpful.

File: pkg/v1/tarball/write.go
Comment (jonjohnsonjr) [inline pkg/v1/tarball/write.go:325]:
```suggestion
	size += calculateSingleFileInTarSize(int64(len(mBytes)))
```

Comment (deitch) [reply pkg/v1/tarball/write.go:325]: Need to update the func def as well, right? I can just update the commit and re-push.

Comment (jonjohnsonjr) [reply pkg/v1/tarball/write.go]: You have `calculateSingleFileInTarSize` in three places and `calculateSingleFileInTarFileSize` in one place. Whichever you prefer, just needs to be consistent. Also feel free to reject my suggestion and just update/rebase whatever 👍

**Comment by jonjohnsonjr** [comment]
Looks like travis doesn't have very many windows machines available :(

**Review by jonjohnsonjr — APPROVED** [review]

**Comment by jonjohnsonjr** [comment]
Travis failed but I didn't notice, retried and it works.

**Pull request merged**
